artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/ActiveMQServerImpl.java

```diff
@@ -3493,12 +3493,25 @@ public void run() {
          recoverStoredBridges();
       }
 
-      if (configuration.getMaxDiskUsage() != -1) {
-         try {
-            injectMonitor(new FileStoreMonitor(getScheduledPool(), executorFactory.getExecutor(), configuration.getDiskScanPeriod(), TimeUnit.MILLISECONDS, configuration.getMaxDiskUsage() / 100f, ioCriticalErrorListener));
-         } catch (Exception e) {
-            ActiveMQServerLogger.LOGGER.unableToInjectMonitor(e);
+      deployFileStoreMonitor();
+   }
+
+   private void deployFileStoreMonitor() throws Exception {
+      FileStoreMonitor.FileStoreMonitorType fileStoreMonitorType = null;
+      Number referenceValue = null;
+      if (configuration.getMinDiskFree() != -1) {
+         if (configuration.getMaxDiskUsage() != -1) {
+            ActiveMQServerLogger.LOGGER.configParamOverride(FileConfigurationParser.MIN_DISK_FREE, configuration.getMinDiskFree(), FileConfigurationParser.MAX_DISK_USAGE, configuration.getMaxDiskUsage());
```

Comment:
This will log a warning, however its clearly documented that minDiskFree takes precedence, and it is already disabled by default in both code and xml config, whereas in contrast maxDiskUsage is enabled by default in both code and xml config. So to avoid a log warning users will *need* to explicitly configure both of them. I'm not sure thats necessary. At the very least it seems more like info or lower level than a warning if we keep it logging.